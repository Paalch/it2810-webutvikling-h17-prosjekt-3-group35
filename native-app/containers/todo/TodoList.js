import React from 'react';
import { StyleSheet, Text, View, Button, TextInput } from 'react-native';
import TodoElement from './TodoElement';

export default class TodoList extends React.Component {
    constructor() {
        super();
        this.state = {
            todos: [],
            text: ''
        };
        this.updateTextField = this.updateTextField.bind(this);
        this.saveTodo = this.saveTodo.bind(this);
        this.removeElement = this.removeElement.bind(this);
    }

    updateTextField(e) {
        this.setState({text: e.target.value});
    }

    removeElement(todo) {
        this.setState({
            todos: this.state.todos.filter((t) =>
                t !== todo
            )
        });
    }

    saveTodo() {
        const todoValue = this.state.text;
        if (todoValue.length === 0) {
            return;
        }
        this.setState({todos: this.state.todos.concat(todoValue), text: ''});
    };

    render() {
        const content = this.state.todos.map((todo, i) =>
            <TodoElement text={todo} key={i} removeElement={this.removeElement}/>
        );
        return (
            <View style={styles.container}>
                <View>
                    <View>
                        <TextInput
                            style={styles.formInput}
                            type={'text'}
                            placeholder={'.............................................................'}
                            value={this.state.text}
                            onChange={this.updateTextField}
                        />
                    </View>
                    <View>
                        <Button onPress={this.saveTodo} title={'+'}/>
                    </View>
                </View>
                <View style={styles.todos}>
                    {content}
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        backgroundColor: "#ccccff",
        padding: 20,
    },

    todos: {
        marginTop: 10,
    },

});
