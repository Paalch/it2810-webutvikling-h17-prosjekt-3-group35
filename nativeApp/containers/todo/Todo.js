import React from 'react';
import StyleSheet from 'react-native';
import Text from 'react-native';
import View from 'react-native';

class Todo extends React.Component {
  render () {
    return (
      <View Style={styles.todoContainer}>
        <View Style={styles.boxHeader}>
          <Text>2do</Text>
        </View>
        <View Style={styles.boxBody}>
          <Text>Lorem ipsum</Text>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
 todoContainer: {
    backgroundColor: "#ccf",
    padding: 30,
    shadowColor: 'rgba(33,33,33,.7)',
     shadowOpacity: 7,
     shadowOffset: {
         height: 5,
         width: 5
     },
 },

 boxheader: {
     flexDirection: 'row'
 },

 boxBocy: {
     flexDirection: 'row'
 }

});

export default Todo;