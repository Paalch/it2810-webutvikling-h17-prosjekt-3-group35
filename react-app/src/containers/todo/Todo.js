import React from 'react';
import './todo.css';

class Todo extends React.Component {
  render () {
    return (
      <div className={'todo-container box'}>
        <h1>Todo</h1>
      </div>
    )
  }
}

export default Todo;