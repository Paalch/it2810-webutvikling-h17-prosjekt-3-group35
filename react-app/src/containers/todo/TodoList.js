import React, {Component} from 'react';
import {Col, Row, FormControl, Button, FormGroup, InputGroup} from 'react-bootstrap';
import TodoElement from "./TodoElement";
import './todo.css';

class TodoList extends Component {

  constructor() {
    super();
    this.state = {
      todos: [],
      text: ''
    };
    this.updateTextField = this.updateTextField.bind(this);
    this.saveTodo = this.saveTodo.bind(this);
    this.removeElement = this.removeElement.bind(this);
    this.save = this.save.bind(this);
    this.load = this.load.bind(this);
  }

  componentDidMount() {
    const data = this.load();
    if (data !== null) {
      this.setState(() => data);
    }
  }

  componentDidUpdate() {
    this.save();
  }

  save() {
    localStorage.setItem('todo', JSON.stringify(this.state));
  }

  load() {
    return 'todo' in localStorage ?
      JSON.parse(localStorage.getItem('todo')) : null;
  }

  updateTextField(e) {
    this.setState({text: e.target.value});
  }

  removeElement(id) {
    this.setState({
      todos: this.state.todos.filter((t) =>
        id !== t.id
      )
    });
  }

  saveTodo() {
    const todoValue = this.state.text;
    if (todoValue.length === 0) {
      return;
    }
    const newTodo = {id: Date.now(), value: todoValue};
    this.setState({todos: this.state.todos.concat(newTodo), text: ''});
  };

  render() {
    const content = this.state.todos.map((element, i) =>
      <TodoElement todo={element} key={i} removeElement={this.removeElement}/>
    );

    return (
      <div>
        <Row>
          <Col xs={10}>
            <FormControl
              id={'todo-form-input'}
              type={'text'}
              placeholder={'.............................................................'}
              value={this.state.text}
              onChange={this.updateTextField}
            />
          </Col>
          <Col xs={2}>
            <Button
              className={'todo-add-button'}
              onClick={this.saveTodo}>
              +
            </Button>
          </Col>
        </Row>
        <div id={'todos'}>
          {content}
        </div>
      </div>
    )
  }
}

export default TodoList;