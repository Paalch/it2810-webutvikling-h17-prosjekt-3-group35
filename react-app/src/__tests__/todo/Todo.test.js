import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Todo from '../../containers/todo/Todo'
import Adapter from 'enzyme-adapter-react-15'
import renderer from 'react-test-renderer'
import TodoList from '../../containers/todo/TodoList'


Enzyme.configure({ adapter: new Adapter() })


describe('Checks that the todo component renders correctly', () => {

    //checks that todo render, by compareing snapshot
    it('Checks that Todo renders', () => {
        const rendered = renderer.create(
            <Todo />
        );
        expect(rendered.toJSON()).toMatchSnapshot()
    });

    //Checks that only one todolist is renderd
    it('Checks that TodoList renders', () => {
        const wrapper = shallow(<Todo />);
        expect(wrapper.find(TodoList)).toHaveLength(1);
    });
});