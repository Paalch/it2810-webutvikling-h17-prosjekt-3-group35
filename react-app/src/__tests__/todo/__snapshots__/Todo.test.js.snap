// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checks that the todo component renders correctly Checks that Todo renders 1`] = `
<div
  className="todo-container box"
>
  <div
    className="todo-header box-header row"
  >
    <h1>
      Todo
    </h1>
  </div>
  <div
    className="todo-body box-body row"
  >
    <div>
      <div
        className="row"
      >
        <div
          className="col-xs-10"
        >
          <input
            className="form-control"
            id="todo-form-input"
            onChange={[Function]}
            placeholder="............................................................."
            type="text"
            value=""
          />
        </div>
        <div
          className="col-xs-2"
        >
          <button
            className="todo-add-button btn btn-default"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </div>
      </div>
      <div
        id="todos"
      />
    </div>
  </div>
</div>
`;
